### PR TITLE
MH-12472, FFmpeg Composer Implementation

### PR DESCRIPTION
**Pull request by lkiesow**
This patch is a major update to the FFmpeg composer implementation
    including several bug fixes as well as a massive reduction in code
    complexity.
    
    - The four abstraction layers for the encoder engines used for the
      single FFmpeg implementation have been collapsed and several unused
      and partly broken methods have been removed or were fixed.
    
    - The encoder engine factory has been removed since all it was doing was
      to instantiate new encoder engines. Something we can easily do directly.
    
    - Several different (and yet mostly the same) methods calling FFmpeg in
      several ways have been unified.
    
    - plexus-utils is now used for parsing command lines from the encoding
      profiles, obsoleting our own buggy parser implementation.
    
    - Lots of tests have been added and slow old tests have been replaced.
    
    But there are still some places to work on:
    
    - The methods `process()` and `parallelEncoder()` in the encoder engine
      basically do the same thing. We can easily replace them (and probably
      a lot of the smaller methods) with a method accepting by creating a
      methods which accepts an arbitrary amount of input and output files.
    
    - The current incident implementation is ugly. They are hard to use and
      make the code a lot worse to read. We should think about modifying
      them, make them easier to use (e.g. have the service registry just
      catch errors and convert them into incidents).
    
    - The encoding profile media type implementation is broken in a lot of
      places. This was discussed on list a while ago. The best solution here
      is probably to just remove them since they do not serve any real
      purpose at this time.

### CODE REVIEW

**Comment by karendolan** [comment]
Hi Lars I'm just downloading to run the sanity checks, but I see that the Travis checks have passed. How do I assign myself as //review//er?

**Comment by lkiesow** [comment]
You should have gotten an invitation to join the committer team. Once, you have joined, you should can assign this pull request to yourself right on this page.

**Review by karendolan — APPROVED** [review]
:thumbsup

**Comment by karendolan** [comment]
How can I tell what place this feature pull is in the current "in order" order? Which method of merge do you recommend: use this UI's merge pull request's "rebase and merge", "create a merge commit" or pull down and merge locally onto develop and push.

**Comment by lkiesow** [comment]
- You can simply [sort the pull requests on Github to show the oldest pull request first](https://github.com/opencast/opencast/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Aopen+base%3Adevelop+sort%3Acreated-asc+). That should work. Of course, the order is also essentially the pull request number.
- There are three pull requests in line before this one. But since they are all from me (and one is not a feature anyway), go ahread and merge this before those pull requests. It should cause no conflicts.


- The merge method you use is up to you. Though it is potentially something we should bring up in the community since we already considered squashing merges at some point. I think most people are using merge commits. I personally prefer merging via command line (also using merge commits) but that's just weird me ;-P
